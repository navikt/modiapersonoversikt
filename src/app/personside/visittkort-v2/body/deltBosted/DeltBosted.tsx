import VisittkortElement from '../VisittkortElement';
import { VisittkortGruppe } from '../VisittkortStyles';
import Infotegn from '../../../../../svg/Info';
import { DeltBosted } from '../../PersondataDomain';
import Adresseinfo from '../AdresseInfo';
import GyldighetsPeriode from '../GyldighetsPeriode';

interface Props {
    deltBosted: DeltBosted[];
}

function DeltBostedElement(props: { deltBosted: DeltBosted }) {
    if (!props.deltBosted.adresse) {
        return null;
    }

    return (
        <VisittkortElement>
            <GyldighetsPeriode gyldighetsPeriode={props.deltBosted.gyldighetsPeriode} />
            <Adresseinfo adresse={props.deltBosted.adresse} />
        </VisittkortElement>
    );
}

function DeltBostedWrapper({ deltBosted }: Props) {
    if (deltBosted.isEmpty()) {
        return null;
    }

    return (
        <VisittkortGruppe tittel={'Delt Bosted'} ikon={<Infotegn />}>
            {deltBosted.map((deltBosted, index) => (
                <DeltBostedElement key={index} deltBosted={deltBosted} />
            ))}
        </VisittkortGruppe>
    );
}

export default DeltBostedWrapper;
