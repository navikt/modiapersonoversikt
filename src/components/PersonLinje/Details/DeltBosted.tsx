import { usePersonData } from 'src/lib/clients/modiapersonoversikt-api';
import type { PersonData } from 'src/lib/types/modiapersonoversikt-api';
import ValidPeriod from '../common/ValidPeriod';
import { Adresseinfo, Group, InfoElement } from './components';

type DeltBosted = PersonData['deltBosted'][0];

function DeltBostedElement({ deltBosted }: { deltBosted: DeltBosted }) {
    if (!deltBosted.adresse) {
        return null;
    }

    return (
        <InfoElement>
            <ValidPeriod
                from={deltBosted.gyldighetsPeriode?.gyldigFraOgMed}
                to={deltBosted.gyldighetsPeriode?.gyldigTilOgMed}
            />
            <Adresseinfo adresse={deltBosted.adresse} />
        </InfoElement>
    );
}

function DeltBosted() {
    const {
        data: { person }
    } = usePersonData();

    const deltBosted = person.deltBosted;

    if (deltBosted.isEmpty()) {
        return null;
    }

    return (
        <Group title="Delt Bosted">
            {deltBosted.map((deltBosted, index) => (
                <DeltBostedElement key={`${deltBosted.adresse?.linje1}-${index}`} deltBosted={deltBosted} />
            ))}
        </Group>
    );
}

export default DeltBosted;
