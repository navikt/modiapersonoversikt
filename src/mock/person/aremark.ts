import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            behovKode: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            telefonnummer: '99887766',
            retningsnummer: '47',
            sistEndretAv: 'BRUKER'
        }
    }
};