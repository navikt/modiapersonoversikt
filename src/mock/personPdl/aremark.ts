import {
    Adresse,
    AdresseBeskyttelse,
    EgenAnsatt,
    Enhet,
    Kjonn,
    KodeBeskrivelse,
    Navn,
    Person,
    PersonStatus,
    Sikkerhetstiltak,
    SikkerhetstiltakType,
    Sivilstand,
    SivilstandType,
    Statsborgerskap,
    Telefon,
    TilrettelagtKommunikasjon
} from '../../models/personPdl/person';
import { getAremarkForeldreansvar } from './foreldreansvarMock';
import { getDeltBostedMock } from './deltBostedMock';
import * as faker from 'faker';
import { getMockDodsbo } from './dodsbo';
import navfaker from 'nav-faker/dist/index';
import { mockVergemal } from './vergemalMock';
import { DigitalKontaktinformasjon } from '../../models/personPdl/digitalKontaktinformasjon';
import { getBankKonto } from './bankkontoMock';
import { Fullmakt, FullmaktsRolle } from '../../models/personPdl/fullmakt';

export const navnAremark: Navn = {
    fornavn: 'TESTFAMILIEN',
    mellomnavn: null,
    etternavn: 'AREMARK'
};

export const kodeBeskrivelseKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.Mann,
    beskrivelse: 'Mann'
};

export const adresseAremark: Adresse = {
    linje1: 'Adresseveien 1, 0661 AREMARKSTAD',
    linje2: null,
    linje3: null
};

export const navEnhetAremark: Enhet = {
    id: '0000',
    navn: 'Nav-kontor'
};

export const statsborgerskapAremark: Statsborgerskap = {
    land: {
        kode: 'NOR',
        beskrivelse: 'Norsk statsborger'
    },
    gyldigFraOgMed: new Date('1980-10-10'),
    gyldigTilOgMed: null
};

export const adressebeskyttelseAremark: KodeBeskrivelse<AdresseBeskyttelse> = {
    kode: AdresseBeskyttelse.Ugradert,
    beskrivelse: 'Ingen'
};

export const sikkerhetstiltakAremark: Sikkerhetstiltak = {
    type: SikkerhetstiltakType.TelefoniskUtestengelse,
    gyldigFraOgMed: new Date('2005-02-13'),
    gyldigTilOgMed: new Date('2020-02-02')
};

export const personstatusAremark: KodeBeskrivelse<PersonStatus> = {
    kode: PersonStatus.Bosatt,
    beskrivelse: 'Bosatt i Norge'
};

export const sivilstandAremark: Sivilstand = {
    type: {
        kode: SivilstandType.Skilt,
        beskrivelse: 'Skilt'
    },
    gyldigFraOgMed: new Date('2010-03-06')
};

export const tilrettelagtKommunikasjonAremark: TilrettelagtKommunikasjon = {
    talesprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ],
    tegnsprak: [
        {
            kode: 'NO',
            beskrivelse: 'Norsk'
        }
    ]
};

export const telefonAremark: Telefon = {
    retningsnummer: {
        kode: '+47',
        beskrivelse: 'retningsnummer er norsk'
    },
    identifikator: '99009900',
    sistEndret: new Date('2017-10-10'),
    sistEndretAv: 'TESTFAMILIEN AREMARK',
    prioritet: 1
};

export const digitalKontaktInformasjonAremark: DigitalKontaktinformasjon = {
    personident: '10108000398',
    reservasjon: 'reservert',
    epostadresse: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: new Date('2013-01-01')
    },
    mobiltelefonnummer: {
        value: null,
        sistOppdatert: null,
        sisVerifisert: null
    }
};

export const fullmaktAremark: Fullmakt = {
    motpartsPersonident: '123456789',
    motpartsPersonNavn: {
        fornavn: 'Navn',
        mellomnavn: null,
        etternavn: 'Navnesen'
    },
    motpartsRolle: FullmaktsRolle.Fullmektig,
    omraade: ['*'],
    gyldigFraOgMed: new Date('2019-01-01'),
    gyldigTilOgMed: new Date('2022-12-12')
};

export const aremark: Person = {
    fnr: '10108000398',
    navn: [navnAremark],
    kjonn: [kodeBeskrivelseKjonn],
    fodselsdato: [new Date('1980-10-10')],
    dodsdato: [],
    bostedAdresse: [adresseAremark],
    kontaktAdresse: [adresseAremark],
    navEnhet: navEnhetAremark,
    statsborgerskap: [statsborgerskapAremark],
    adressebeskyttelse: [adressebeskyttelseAremark],
    sikkerhetstiltak: [sikkerhetstiltakAremark],
    erEgenAnsatt: EgenAnsatt.Nei,
    personstatus: [personstatusAremark],
    sivilstand: [sivilstandAremark],
    foreldreansvar: [getAremarkForeldreansvar()],
    deltBosted: getDeltBostedMock(faker),
    dodsbo: getMockDodsbo(faker),
    fullmakt: [fullmaktAremark],
    vergemal: mockVergemal(navfaker.personIdentifikator.f√∏dselsnummer()),
    tilrettelagtKommunikasjon: tilrettelagtKommunikasjonAremark,
    telefonnummer: [telefonAremark],
    kontaktOgReservasjon: digitalKontaktInformasjonAremark,
    bankkonto: getBankKonto()
};
