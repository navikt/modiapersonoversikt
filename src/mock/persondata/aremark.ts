import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';
import { harDiskresjonskode } from '../../app/personside/visittkort-v2/visittkort-utils';

export function hentBarnAremark(): ForelderBarnRelasjon[] {
    return [
        {
            ident: '12345678910',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: 13,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [
                {
                    kode: AdresseBeskyttelse.UGRADERT,
                    beskrivelse: 'UGRADERT'
                }
            ],
            harSammeAdresse: false,
            dodsdato: []
        },
        {
            ident: '12345678911',
            rolle: ForelderBarnRelasjonRolle.BARN,
            navn: [
                {
                    fornavn: 'BARN1',
                    mellomnavn: null,
                    etternavn: 'BARNESEN'
                }
            ],
            fodselsdato: ['2008-03-15' as LocalDate],
            alder: null,
            kjonn: [
                {
                    kode: Kjonn.K,
                    beskrivelse: 'Kvinne'
                }
            ],
            adressebeskyttelse: [],
            harSammeAdresse: false,
            dodsdato: []
        }
    ];
}

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    geografiskTilknytning: '0118',
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge',
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2010-02-20T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'FREG'
            },
            gyldighetsPeriode: null
        }
    ],
    oppholdsAdresse: [
        {
            linje1: 'Oppholdsadresse 1',
            linje2: '0000 AREMARK',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum',
        publikumsmottak: [
            {
                besoksadresse: {
                    linje1: 'Adressevei 1',
                    linje2: '0000 AREMARK',
                    linje3: null,
                    sistEndret: null,
                    gyldighetsPeriode: null
                },
                apningstider: [
                    {
                        ukedag: 'Mandag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Tirsdag',
                        apningstid: '09.00 - 15.00'
                    },
                    {
                        ukedag: 'Onsdag',
                        apningstid: '10.00 - 16.00'
                    }
                ]
            }
        ]
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Kode 6'
        }
    ],
    sikkerhetstiltak: [
        {
            type: 'TFUS',
            beskrivelse: 'Telefonisk utestengelse',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate,
            sivilstandRelasjon: {
                fnr: '12345555555',
                navn: [
                    {
                        fornavn: 'BAREMARK',
                        mellomnavn: null,
                        etternavn: 'TESTFAMILIEN'
                    }
                ],
                alder: 40,
                adressebeskyttelse: [
                    {
                        kode: AdresseBeskyttelse.UGRADERT,
                        beskrivelse: 'UGRADERT'
                    }
                ],
                harSammeAdresse: true
            }
        }
    ],
    foreldreansvar: hentBarnAremark().map((barn) => ({
        ansvar: 'felles',
        ansvarlig: null,
        ansvarsubject: {
            navn: harDiskresjonskode(barn.adressebeskyttelse) ? null : barn.navn.firstOrNull(),
            ident: barn.ident
        }
    })),
    deltBosted: [
        {
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            },
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null,
                sistEndret: null,
                gyldighetsPeriode: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG,
            sistEndret: {
                ident: 'Folkeregisteret',
                tidspunkt: '2020-01-01T10:15:30' as LocalDateTime,
                system: 'Folkeregisteret'
            }
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omrade: [
                {
                    kode: '*',
                    beskrivelse: ''
                }
            ],
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'Voksen',
            omfang: 'Ivareta personens interesser innenfor det personlige og økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-06-01' as LocalDate,
                gyldigTilOgMed: '2022-05-10' as LocalDate
            }
        },
        {
            ident: '123456799',
            navn: {
                fornavn: 'Truls',
                mellomnavn: null,
                etternavn: 'Tøffel'
            },
            vergesakstype: 'Fremtidsfullmakt',
            omfang: 'Ivareta personens interesser innenfor det økonomiske området',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetsPeriode: {
                gyldigFraOgMed: '2021-02-10' as LocalDate,
                gyldigTilOgMed: '2022-05-01' as LocalDate
            }
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'Norge'
            },
            identifikator: '99009900',
            sistEndret: {
                ident: '11223344',
                tidspunkt: '2018-06-14T00:00:00' as LocalDateTime,
                system: 'BD06'
            },
            prioritet: -1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '12345678',
            sistOppdatert: '2021-11-02' as LocalDate,
            sistVerifisert: '2021-11-02' as LocalDate
        }
    },
    bankkonto: {
        kontonummer: '12345678900',
        banknavn: 'DNB ASA',
        sistEndret: {
            ident: '1010800 BD03',
            tidspunkt: '2006-03-15T00:00:00' as LocalDateTime,
            system: ''
        },
        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null,
            sistEndret: null,
            gyldighetsPeriode: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: hentBarnAremark()
};
