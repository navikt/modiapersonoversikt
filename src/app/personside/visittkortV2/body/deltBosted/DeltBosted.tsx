import { Adresse } from '../../../../../models/personPdl/person';
import { DeltBosted } from '../../../../../models/personPdl/deltBosted';
import { Normaltekst } from 'nav-frontend-typografi';
import React from 'react';
import { formaterDato } from '../../../../../utils/string-utils';
import VisittkortElement from '../../../visittkort/body/VisittkortElement';
import { VisittkortGruppe } from '../../../visittkort/body/VisittkortStyles';
import Infotegn from '../../../../../svg/Info';

function harAdresse(adresse: Adresse | null) {
    return adresse !== undefined;
}

function getAdresse(deltBosted: DeltBosted) {
    if (!deltBosted?.adresse) {
        return null;
    }

    const { adresse } = deltBosted;

    const gateadresse = `${adresse.linje1} ${adresse?.linje2} ${adresse?.linje3}`;

    return (
        <div key={adresse.linje1}>
            <Normaltekst>{gateadresse}</Normaltekst>
        </div>
    );
}

function DeltBostedElement(props: { deltBosted: DeltBosted }) {
    const startDato = props.deltBosted.startdatoForKontrakt
        ? formaterDato(props.deltBosted.startdatoForKontrakt)
        : 'dato ikke funnet';
    const sluttDato = props.deltBosted.sluttdatoForKontrakt
        ? formaterDato(props.deltBosted.sluttdatoForKontrakt)
        : 'dato ikke funnet';
    const ukjentBosted = !harAdresse(props.deltBosted.adresse) ? <Normaltekst>Ukjent bosted</Normaltekst> : undefined;

    return (
        <VisittkortElement>
            <Normaltekst>{getAdresse(props.deltBosted) ?? ukjentBosted}</Normaltekst>
            <Normaltekst>Gyldig fra og med {startDato}</Normaltekst>
            <Normaltekst>Gyldig til og med {sluttDato}</Normaltekst>
        </VisittkortElement>
    );
}

function DeltBostedWrapper(props: { deltbosted: DeltBosted[] }) {
    if (!props.deltbosted || props.deltbosted.length == 0) {
        return null;
    }

    const delteBosteder = props.deltbosted.map((deltbosted, index) => (
        <DeltBostedElement key={index} deltBosted={deltbosted} />
    ));

    return (
        <VisittkortGruppe tittel={'Delt Bosted'} ikon={<Infotegn />}>
            {delteBosteder}
        </VisittkortGruppe>
    );
}

export default DeltBostedWrapper;
