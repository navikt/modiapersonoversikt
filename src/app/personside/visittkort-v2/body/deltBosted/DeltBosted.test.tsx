import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../../test/Testprovider';
import DeltBosted from './DeltBosted';
import { aremark } from '../../../../../mock/persondata/aremark';

test('viser deltbosted', () => {
    const visittkortbody = renderer.create(
        <TestProvider>
            <DeltBosted deltBosted={aremark.deltBosted} />
        </TestProvider>
    );

    expect(visittkortbody.toJSON()).toMatchSnapshot();
});
