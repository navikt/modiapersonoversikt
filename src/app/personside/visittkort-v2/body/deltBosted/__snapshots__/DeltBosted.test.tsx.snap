// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`viser deltbosted 1`] = `
.c0 {
  position: relative;
  display: flex;
  align-items: center;
}

.c1 {
  position: absolute;
  left: -3.125rem;
  width: 3.125rem;
  display: flex;
  justify-content: center;
}

.c1 >svg {
  height: 1.5rem;
  width: auto;
}

.c3 {
  margin: 10px 0 20px 0;
}

.c3:last-child {
  margin: 10px 0 0 0;
}

.c2 {
  margin-bottom: 1rem;
}

.c4 {
  color: #645f5a;
  margin-top: 3px;
}

.c4 .typo-etikett-liten {
  line-height: 1rem;
}

<section>
  <div
    className="c0"
  >
    <div
      className="c1"
    >
      <svg
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0zM9 19v-2h2v-5H9v-2h4v7h2v2H9zm3-14a1.5 1.5 0 110 3 1.5 1.5 0 010-3z"
          fill="#0067C5"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <h3
      className="typo-undertittel"
    >
      Delt Bosted
    </h3>
  </div>
  <div
    className="c2"
  />
  <div
    className="c3"
  >
    <div
      className="c0"
    />
    <div
      className="c4"
    >
      <p
        className="typo-undertekst"
      >
        Gyldig: 
        01.02.2021 â€“ 01.05.2022
      </p>
    </div>
    <p
      className="typo-normal"
    >
      Adresseveien 1
    </p>
    <p
      className="typo-normal"
    >
      0000 Aremark
    </p>
  </div>
</section>
`;
